caffe2/share/contrib/observers/observer.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
```

Comment:
This file should be named something that exposes the class name.  there already exists an `observer.h` in `caffe2/core` so it gets confusing to have this

Reply:
How about call it time_net_observer.h?